source/dem/find_contact_detection_step.cc

```diff
@@ -4,32 +4,29 @@ using namespace dealii;
 
 template <int dim>
 bool
-find_contact_detection_step(
-  Particles::ParticleHandler<dim> &particle_handler,
-  const double &                   dt,
-  const double &                   smallest_contact_search_criterion,
-  MPI_Comm &                       mpi_communicator,
-  bool &                           sorting_in_subdomains_step,
-  std::unordered_map<types::particle_index, double> &displacement)
+find_contact_detection_step(Particles::ParticleHandler<dim> &particle_handler,
+                            const double &                   dt,
+                            const double &smallest_contact_search_criterion,
+                            MPI_Comm &    mpi_communicator,
+                            bool &        sorting_in_subdomains_step,
+                            std::vector<double> &displacement)
 {
-  double       max_displacement       = 0;
-  unsigned int contact_detection_step = 0;
-
-  // Looping through all the particles:
   if (sorting_in_subdomains_step)
-    {
-      // If last step was a sorting into subdomains step, the displcement
-      // is reinitialized and then the new displacement is calculated
+    for (auto &d : displacement)
+      d = 0.;
 
-      // Clearing displacement
-      displacement.clear();
-    }
+  double       max_displacement       = 0;
+  unsigned int contact_detection_step = 0;
 
   // Updating displacement
   for (auto &particle : particle_handler)
     {
-      auto &particle_properties   = particle.get_properties();
+      auto &particle_properties = particle.get_properties();
+#if DEAL_II_VERSION_GTE(10, 0, 0)
+      auto &particle_displacement = displacement[particle.get_local_index()];
```

Comment:
This is the main reason I've used unordered_map instead of vector. When the container size becomes large, finding an element is faster in maps. Do you think using vectors is a suitable choice?

Reply:
There is no use of std::find() here so there is no searching. Only the [] operator is used. Consequently, the cost here should be lower and lower as the number of particles decreases. The reason why we used an unordered map in the past was that we did not have any way to number the local particles, hence we had to access them through the ID which is not a continuous numbering scheme.